io-engine/src/lvs/lvol_snapshot.rs

```diff
@@ -35,41 +35,51 @@ use strum::{EnumCount, IntoEnumIterator};
 use super::Error;
 use futures::future::join_all;
 
-#[async_trait(?Send)]
-trait AsyncIterator {
+pub trait AsyncIterator {
     type Item;
-    async fn next(&mut self) -> Option<SnapshotParams>;
+    fn parent(&mut self) -> Option<Self::Item>;
```

Comment:
I think should keep next which is idiomatic in rust

```suggestion
    fn next(&mut self) -> Option<Self::Item>;
```

Reply:
We have another API in SPDK to get the next blob. Reserve that name when We will use `bs_iter_next` function which is internally calling `spdk_bs_iter_next` function.

Reply:
Then maybe rename this iterator to something like `AsyncParentIterator` ?